### PR TITLE
Allow empty transactionHash from logs of parity light client

### PR DESCRIPTION
**Pull request by dimroc**
This introduces `gencodec` as a go generator to produce
the json serialization logic of `store.Log` in `gen_log_json.go`.

After installing `gencodec` with:

```bash
go get -u github.com/smartcontractkit/gencodec
```

Simply running `go generate ./...` will produce the file.

### CODE REVIEW

**Comment by thodges-gh** [comment]
Might be helpful, I receive the following when trying to build with `make install`:

```
store/gen_log_json.go:9:2: must be imported as github.com/ethereum/go-ethereum/common
store/gen_log_json.go:10:2: must be imported as github.com/ethereum/go-ethereum/common/hexutil
Makefile:40: recipe for target 'install' failed
make: *** [install] Error 1
```

The two lines mentioned in `store/gen_log_json.go`:
```
	"github.com/smartcontractkit/chainlink/vendor/github.com/ethereum/go-ethereum/common"
	"github.com/smartcontractkit/chainlink/vendor/github.com/ethereum/go-ethereum/common/hexutil"
```

**Comment by dimroc** [comment]
@thodges-gh Can you see if it works with

```bash
go get -u github.com/smartcontractkit/gencodec
```
?

I also worry about the implications this has on our `make install` step. Do we now require `go get -u github.com/smartcontractkit/gencodec` in our installation instructions for non devs?

**Comment by thodges-gh** [comment]
@dimroc I did run `go get -u github.com/smartcontractkit/gencodec` before `make install` and the error I pasted before was the result.

File: store/types.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	Topics []common.Hash `json:"topics" gencodec:"required"`
+	// supplied by the contract, usually ABI-encoded
+	Data []byte `json:"data" gencodec:"required"`
+
+	// Derived fields. These fields are filled in by the node
+	// but not secured by consensus.
+	// block in which the transaction was included
+	BlockNumber uint64 `json:"blockNumber"`
+	// hash of the transaction
+	TxHash common.Hash `json:"transactionHash"`
```
Comment (dimroc) [inline store/types.go]: `TxHash` is now no longer `required`, unlike [go-ethereum's](https://github.com/ethereum/go-ethereum/blob/31c4e3a118b0bc43e3340e565d75160afadd2438/core/types/log.go#L45).

**Comment by dimroc** [comment]
Example payload from parity's light client. The missing `transactionHash` is what caused the this bug.

```
transactionHash:<nil>
transactionIndex:0x3
address:0x1aee7c03606fca5035d204c3818d0660bb230e44
blockNumber:0x8bf99b
removed:false
topics:[something]
type:mined
blockHash:0xdb777676330c067e3c3a6dbfc2d51282cac5bcc1b7a884dd8d85ba72ca1f147e
data:something
logIndex:0x5
transactionLogIndex:0x3
```

**Pull request merged**